frontend/components/user-settings/user-settings.tsx

```diff
@@ -291,21 +355,106 @@ export default function UserSettings({ userId }: { userId: string }) {
+                  <Input
+                    id="newPassword"
+                    type={showNewPassword ? "text" : "password"}
+                    className={cn("pr-10", !passwordsMatch && "border-red-500")}
+                    value={newPassword}
+                    onChange={(e) => setNewPassword(e.target.value)}
+                  />
+                  <Button
+                    type="button"
+                    variant="ghost"
+                    size="icon"
+                    className="absolute inset-y-0 right-0 h-full px-3 flex items-center"
+                    onClick={() => togglePasswordVisibility("new")}
+                  >
+                    {showNewPassword ? (
+                      <EyeOff className="h-4 w-4" />
+                    ) : (
+                      <Eye className="h-4 w-4" />
+                    )}
+                  </Button>
+                </div>
               </div>
-              <div>
+              <div className="relative">
                 <Label htmlFor="confirmPassword">Confirm New Password</Label>
-                <Input id="confirmPassword" type="password" />
+                <div className="relative">
+                  <Input
+                    id="confirmPassword"
+                    type={showConfirmPassword ? "text" : "password"}
+                    className={cn("pr-10", !passwordsMatch && "border-red-500")}
+                    value={confirmPassword}
+                    onChange={(e) => setConfirmPassword(e.target.value)}
+                  />
+                  <Button
+                    type="button"
+                    variant="ghost"
+                    size="icon"
+                    className="absolute inset-y-0 right-0 h-full px-3 flex items-center"
+                    onClick={() => togglePasswordVisibility("confirm")}
+                  >
+                    {showConfirmPassword ? (
+                      <EyeOff className="h-4 w-4" />
+                    ) : (
+                      <Eye className="h-4 w-4" />
+                    )}
+                  </Button>
+                </div>
+                {!passwordsMatch && (
+                  <p className="text-sm text-red-500 mt-1">
+                    Passwords do not match
+                  </p>
+                )}
+                {!isPasswordValid && newPassword && (
+                  <p className="text-sm text-red-500 mt-1">
+                    Password must be at least 8 characters long, include 1
+                    uppercase letter and 1 special character.
+                  </p>
+                )}
               </div>
             </CardContent>
             <CardFooter>
-              <Button>Change Password</Button>
+              <Button
+                disabled={
+                  !passwordsMatch ||
+                  !isPasswordValid ||
+                  !newPassword ||
+                  !confirmPassword ||
+                  !currentPassword
```

Comment:
Can we make another variable to be a combination of everything?

something like disableChangePassword =  !passwordsMatch ||  !isPasswordValid ||                   !newPassword ||                  !confirmPassword ||                  !currentPassword